designs/2024-report-unused-inline-configs/README.md

```diff
@@ -0,0 +1,253 @@
+How a specific language computes its configuration comments is irrelevant to this proposed feature.
+
+### Legacy ("eslintrc") Configs
+
+ESLint v9 is released and flat configs are the default for users.
+The legacy config system will likely be removed in the next major version of ESLint.
+This RFC prefers to avoid the added maintenance cost of supporting the legacy config system.
+
+If this RFC's new CLI flag or config file entry are enabled with using a legacy config, ESLint should throw an error.
+
+## Alternatives
+
+### Checking Config File Values
+
+[eslint/eslint#15476 Change Request: report unnecessary config overrides](https://github.com/eslint/eslint/issues/15476) previously suggested also checking config files (`eslint.config.*`).
+Doing so could be beneficial to flag values that become unnecessary in config files over time.
+However, because flat config purely involves spreading objects, there's no way to know what objects originate from shared configs or shared packages.
+
+[c339a5](https://github.com/eslint/rfcs/pull/121/commits/c339a59bfaec0c48817a012c2f5e92a242a1b1e6) is a reference commit of how this RFC might look with that support added in.
+
+An alternative could have been to require users provide metadata along with their shared configs, and/or wrap them in some function provided by ESLint.
+That could be a future requirement opted into ESLint.
+It's too late now to add that to the flat config system, and as such is out of scope for this RFC.
+
+### Separate CLI Option
+
+Unlike the changes discussed in [Change Request: Enable reportUnusedDisableDirectives config option by default](https://github.com/eslint/eslint/issues/15466) -> [feat!: flexible config + default reporting of unused disable directives](https://github.com/eslint/rfcs/pull/100), reporting unused inline configs does not have legacy behavior to keep to.
+The existing `--report-unused-disable-directives` (enabling) and `--report-unused-disable-directives-severity` (severity) options were kept separate for backwards compatibility.
+
+Adding a sole `--report-unused-inline-configs` CLI option presents a discrepency between the two sets of options.
+An alternative could be to instead add `--report-unused-inline-configs` and `--report-unused-inline-configs-severity` options for consistency's sake.
+
+This RFC's opinion is that the consistency of adding two new options is not worth the excess options logic.
+
+### Superset Behavior: Unused Disable Directive Reporting
+
+Disable directives can be thought as a subset of inline configs in general.
+Reporting on unused disable directives could be thought of as a subset of reporting on unused inline configs.
+
+An additional direction this RFC could propose would be to have the new unused inline config reporting act as a superset of unused disable directive reporting.
+
+However, deprecating `reportUnusedDisableDirectives` would be a disruptive breaking change.
```

Comment:
I do like the idea! 👍 

deprecating is not a breaking change: it's breaking to remove its support. :)

Reply:
I also like this idea and agree that it's not the right time to make such a change.

Reply:
Rephrased to:

> disruptive change and eventual disruptive breaking change.

I'm not confident in that phrasing.